### PR TITLE
fix(file): exit 13 on os error 13

### PR DESCRIPTION
**Pull request by cafkafk**
Signed-off-by: Christina Sørensen <christina@cafkafk.com>

Closes: #317

### CODE REVIEW

File: src/fs/file.rs [hunk 1]
```diff
@@ -204,6 +204,7 @@ impl<'dir> File<'dir> {
     /// Returns an IO error upon failure, but this shouldn’t be used to check
     /// if a `File` is a directory or not! For that, just use `is_directory()`.
     pub fn to_dir(&self) -> io::Result<Dir> {
+        trace!("to_dir reading dir");
```
Comment (MartinFillon) [inline src/fs/file.rs:207]: Just a small display change
```suggestion
        trace!("to_dir: reading dir");
```

File: src/main.rs [hunk 2]
```diff
@@ -378,4 +386,7 @@ mod exits {
 
     /// Exit code for when the command-line options are invalid.
     pub const OPTIONS_ERROR: i32 = 3;
+
+    /// Exit code for when the you don't have permissions for the file.
```
Comment (PThorpe92) [inline src/main.rs:390]:
```suggestion
    /// Exit code for missing file permissions
```

File: src/main.rs [hunk 3]
```diff
@@ -118,8 +117,11 @@ fn main() {
             let theme = options.theme.to_theme(terminal_size::terminal_size().is_some());
             let exa = Exa { options, writer, input_paths, theme, console_width, git };
 
+
+            info!("matching on exa.run");
             match exa.run() {
                 Ok(exit_status) => {
+                    trace!("exa.run exit Ok(exit_status)");
```
Comment (MartinFillon) [inline src/main.rs:124]:
```suggestion
                    trace!("exa.run: exit Ok(exit_status)");
```

File: src/main.rs [hunk 4]
```diff
@@ -130,6 +132,7 @@ fn main() {
 
                 Err(e) => {
                     eprintln!("{e}");
+                    trace!("exa.run exit RUNTIME_ERROR");
```
Comment (MartinFillon) [inline src/main.rs:135]:
```suggestion
                    trace!("exa.run: exit RUNTIME_ERROR");
```

File: src/fs/dir.rs [hunk 5]
```diff
@@ -42,6 +42,7 @@ impl Dir {
                           .map(|result| result.map(|entry| entry.path()))
                           .collect::<Result<_, _>>()?;
 
+        info!("Read directory succes {:?}", &path);
```
Comment (cafkafk) [inline src/fs/dir.rs:45]:
```suggestion
        info!("Read directory success {:?}", &path);
```

**Comment by cafkafk** [comment]
@asahi-m do you think exiting with `13` makes sense, or what are you expecting?

**Pull request merged**